src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -147,14 +150,27 @@ void HexagonGame::draw()
 
     if(Config::get3D())
     {
-        const float depth(styleData._3dDepth);
+        const float aboveMain(-styleData._3dLayerOffset - 1);
+        const bool renderAbove(aboveMain > 0 && !styleData._3dMainOnTop);
+        const float layerOffset(renderAbove ? -1.f : styleData._3dLayerOffset);
+        const float depth(styleData._3dDepth - renderAbove * aboveMain);
         const std::size_t numWallQuads(wallQuads.size());
         const std::size_t numPivotQuads(pivotQuads.size());
         const std::size_t numPlayerTris(playerTris.size());
 
-        wallQuads3D.reserve(numWallQuads * depth);
-        pivotQuads3D.reserve(numPivotQuads * depth);
-        playerTris3D.reserve(numPlayerTris * depth);
+        if(renderAbove)
+        {
+            wallQuads3DTop.reserve(numWallQuads * aboveMain);
+            pivotQuads3DTop.reserve(numPivotQuads * aboveMain);
+            playerTris3DTop.reserve(numPlayerTris * aboveMain);
+        }
+
+        if(depth > 0)
```

Comment:
Would be nice to have a comment explaining why this only happens when `depth > 0`.